lib/galaxy/model/store/__init__.py

```diff
@@ -1407,9 +1415,11 @@ def add_job_output_dataset_associations(self, job_id, name, dataset_instance):
             job_output_dataset_associations[job_id] = {}
         job_output_dataset_associations[job_id][name] = dataset_instance
 
-    def add_dataset_collection(self, collection):
+    def add_dataset_collection(self, collection: model.HistoryDatasetCollectionAssociation, include_files=False):
         self.collections_attrs.append(collection)
         self.included_collections.append(collection)
+        for dataset in collection.collection.dataset_instances:
+            self.add_dataset(dataset, include_files=include_files)
```

Comment:
I have a variant of this called export_collection in the deferred data branch.

Reply:
Awesome, I copied that into https://github.com/galaxyproject/galaxy/pull/12459/commits/2c64e135eaf054d41f6797772286339487dfea7c